.github/workflows/tests.yaml

```diff
@@ -34,6 +34,12 @@ jobs:
           juju-channel: 3.1
           bootstrap-options: '--agent-version=3.1.0'
 
+      # Workaround for https://github.com/canonical/charmcraft/issues/907
+      - name: Change permissions touch the charm file to modify mtime
+        run: |
+          chmod u+x src/charm.py
+          touch src/charm.py
+
```

Comment:
If you add the necessary permissions to `src/charm.py` and push them wouldn't this be solved?